test/ExecutionFramework.cpp

```diff
@@ -170,6 +174,27 @@ void ExecutionFramework::sendMessage(bytes const& _data, bool _isCreation, u256
 		m_transactionSuccessful = (m_gas != m_gasUsed);
 }
 
+void ExecutionFramework::waitForTransaction(std::string const& _txHash) const
+{
+	for (int polls = 0; polls < 3000; polls++)
+	{
+		std::this_thread::sleep_for(std::chrono::milliseconds(1));
+		auto pendingBlock = m_rpc.eth_getBlockByNumber("pending", false);
+
+		if (!pendingBlock["transactions"].empty())
+		{
+			BOOST_REQUIRE_EQUAL(pendingBlock["transactions"][0].asString(), _txHash);
+			return;
+		}
+
+		if (polls == 200)
+		{
+			cerr << "Note: Already used 200 iterations while waiting for transaction confirmation. Issuing an eth_flush request." << endl;
+			m_rpc.rpcCall("eth_flush");
+		}
+	}
+}
```

Comment:
Actually, I expected a ``BOOST_FAIL("Transaction not there after 3000 polls");`` or something at the end of the function :-) - but actually, we might as well just return and hope for the best :-) - worst thing is that we get the RPC error later.

Reply:
exactly my thoughts. It might as well work anyway.